tests/unit_tests/test_suassystem_sda_converter.py

```diff
@@ -27,6 +27,63 @@ def setup_client_converter(self):
 
         self.sda_converter = SDAConverter(self.initial_coordinates, self.boundary_points)
 
+    def test_convert_fly_zones(self):
+
+        result_from_the_program=self.sda_converter.convert_fly_zones(numpy.array([self.boundary_points]))
+        test_location1=Location(self.boundary_points[0]["boundary_pts"][0]["latitude"],self.boundary_points[0]["boundary_pts"][0]["longitude"],0)
+        expected_result=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_result[0],result_from_the_program[0][0][0])
+        self.assertEqual(expected_result[1],result_from_the_program[0][0][1])
+
+    def test_convert_fly_zone(self):
+
+        result_fly_zone=self.sda_converter.convert_fly_zone(self.boundary_points)
+        test_location1=Location(self.boundary_points[0]["boundary_pts"][0]["latitude"],self.boundary_points[0]["boundary_pts"][0]["longitude"],0)
+        expected_fly_zone=(convert_to_point(self.initial_coordinates,test_location1)[:2])
+        self.assertEqual(expected_fly_zone[0],result_fly_zone[0][0])
+        self.assertEqual(expected_fly_zone[1],result_fly_zone[0][1])
+
+    def test_is_obstacel_in_path(self):
+        test_waypoint = Location(38.8705508017, -77.3210866223, 91.44000244140625)
+        self.sda_converter.add_waypoint(test_waypoint)
+        new_obstacle = interop.StationaryObstacle(38.8704274509, -77.3210866223, 60, 91.44000244140625)
+        self.sda_converter.add_obstacle(Location(38.8704274509, -77.3210866223, 91.44000244140625), new_obstacle)
+        self.assertEqual(False,self.sda_converter.is_obstacle_in_path())
+
+
+    def test_does_guided_path_exist(self):
+        self.sda_converter.set_guided_path(numpy.array([numpy.array([1,2,3]),numpy.array([2,2,3])]))
+        self.assertTrue(self.sda_converter.does_guided_path_exist())
+       
+    def test_avoid_obstacles(self):
+        self.sda_converter.avoid_obstacles()
+       
+    def test_has_path_changed(self):
+        path1=numpy.array([numpy.array([1,2,3]),numpy.array([2,2,3])])
+        path2=numpy.array([numpy.array([1,2,3]),numpy.array([2,2,3]),numpy.array([3,2,3])])
+
+        self.assertTrue(self.sda_converter.has_path_changed(path1,path2))
```

Comment:
You need an additional test here for when the number of points in the paths have not changed, but one of the points is different.